feat: add prefetch argument to client post method (#5607)
jina-ai/jina: docs/concepts/client/rate-limit.md

(client-post-prefetch)=
# Rate Limit

There are two ways of applying a rate limit using the {class}`~jina.Client`. 
1. Set using the `Client` class constructor and defaults to 1000 requests. 
1. Set the argument when using {meth}`~jina.clients.mixin.PostMixin.post` method. If not provided, the deafult value of

> 1000 requests will be used. The method argument will override the argument provided in the `Client` class constructor.

Reviewer: ```suggestion
1,000 requests will be used. The method argument will override the argument provided in the `Client` class constructor.
```
Use commas to separate thousands in numbers (unless in code blocks)

Reviewer: Need to fix throughout PR

The `prefetch` argument controls the number of in flight requests made by the {meth}`~jina.clients.mixin.PostMixin.post` 
method. Using the default value might overload the {class}`~jina.Flow` especially if the operation characteristics of the `Flow` 

> are unknown. Furthermore the Client can send various types requests which can have varying resource usage in the `Flow`.

Reviewer: ```suggestion
are unknown. Furthermore the Client can send various types of requests which can have varying resource usage in the `Flow`.
```

For example, a high number of `index` requests can contain a large data payload requiring high input/output operation.
This increases CPU consumption and eventually lead to a build up of the requests on the Flow. If during 

> the queue of in flight requests is already large, a very light weight `search` request to return the total number of 

Reviewer: > If during the queue of in flight requests is already large

I don't understand this. Strange grammar

> documents in the index might be blocked until the queue of `index` requests can be completely processed. To prevent such a scenario,

Reviewer: ```suggestion
Documents in the index might be blocked until the queue of `index` requests can be completely processed. To prevent such a scenario,
```
Capitalize "Documents" throughout

apply the `prefetch` value on the {meth}`~jina.clients.mixin.PostMixin.post` method to limit the rate of
requests for expensive operations.

Apply the `prefetch` argument on the {meth}`~jina.clients.mixin.PostMixin.post` method to dynamically increase 

> the `Flow` responsiveness for customer facing requests which require faster response times vs. background requests such as cronjobs or 

Reviewer: ```suggestion
the `Flow` responsiveness for customer-facing requests which require faster response times vs. background requests such as cronjobs or 
```

# Rate Limit

There are two ways of applying a rate limit using the {class}`~jina.Client`. 
1. Set using the `Client` class constructor and defaults to 1,000 requests. 

> 1. Set the argument when using {meth}`~jina.clients.mixin.PostMixin.post` method. If not provided, the deafult value of

Reviewer: ```suggestion
1. Set the argument when using {meth}`~jina.clients.mixin.PostMixin.post` method. If not provided, the default value of
```

1,000 requests will be used. The method argument will override the argument provided in the `Client` class constructor.

The `prefetch` argument controls the number of in flight requests made by the {meth}`~jina.clients.mixin.PostMixin.post` 
method. Using the default value might overload the {class}`~jina.Flow` especially if the operation characteristics of the `Flow` 
are unknown. Furthermore the Client can send various types of requests which can have varying resource usage in the `Flow`.

For example, a high number of `index` requests can contain a large data payload requiring high input/output operation.

> This increases CPU consumption and eventually lead to a build up of the requests on the Flow. If the queue of in flight requests 

Reviewer: ```suggestion
This increases CPU consumption and eventually lead to a build up of the requests on the Flow. If the queue of in-flight requests 
```
